addons/mail/models/mail_channel.py

```diff
@@ -89,21 +89,31 @@ def _generate_random_token(self):
         required=True, default='groups',
         help='This group is visible by non members. Invisible groups can add members through the invite button.')
     group_public_id = fields.Many2one('res.groups', string='Authorized Group',
-                                      default=lambda self: self.env.ref('base.group_user'))
+                                      compute='_compute_group_public_id', readonly=False, store=True)
 
     _sql_constraints = [
         ('uuid_unique', 'UNIQUE(uuid)', 'The channel UUID must be unique'),
     ]
 
-    # CHAT CONSTRAINT
+    # CONSTRAINTS
 
     @api.constrains('channel_last_seen_partner_ids', 'channel_partner_ids')
     def _constraint_partners_chat(self):
         for ch in self.sudo().filtered(lambda ch: ch.channel_type == 'chat'):
             if len(ch.channel_last_seen_partner_ids) > 2 or len(ch.channel_partner_ids) > 2:
                 raise ValidationError(_("A channel of type 'chat' cannot have more than two users."))
 
+    @api.constrains('group_public_id', 'group_ids')
+    def _constraint_group_id_chat(self):
+        for ch in self.sudo().filtered(lambda ch: ch.channel_type != 'channel'):
+            if ch.group_public_id or ch.group_ids:
+                raise ValidationError(_("A channel of type 'chat'/'group' should not have constraints on the public group settings."))
+
     # COMPUTE / INVERSE
+    @api.depends('channel_type')
+    def _compute_group_public_id(self):
```

Comment:
guideline: have the compute method sorted among them the same way as the corresponding fields

(also keep the empty line below the # title)